File: check_braces.js
import fs from 'fs';

function checkBrackets(filename) {
    const content = fs.readFileSync(filename, 'utf8');
    let stack = [];
    let line = 1;
    let col = 1;
    let inString = null;
    let inComment = false;

    for (let i = 0; i < content.length; i++) {
        let char = content[i];
        let nextChar = content[i+1];

        if (char === '\n') {
            line++;
            col = 1;
            inComment = false;
        } else {
            col++;
        }

        if (inComment) continue;
        if (inString) {
            if (char === inString && content[i-1] !== '\\') inString = null;
            continue;
        }

        if (char === '/' && nextChar === '/') {
            inComment = true;
            i++;
            continue;
        }

        if (char === "'" || char === '"' || char === '`') {
            inString = char;
            continue;
        }

        if (char === '{') {
            stack.push({char: '{', line, col});
        } else if (char === '}') {
            if (stack.length === 0) {
                console.log(`Extra '}' found at line ${line}, col ${col}`);
                return;
            }
            stack.pop();
        }
    }

    if (stack.length > 0) {
        stack.forEach(s => console.log(`Unclosed '${s.char}' from line ${s.line}, col ${s.col}`));
    } else {
        console.log("Brackets are balanced!");
    }
}

checkBrackets(process.argv[2]);

File: scripts/check_braces.py
import sys

def check_brackets(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    stack = []
    line = 1
    col = 1
    for i, char in enumerate(content):
        if char == '\n':
            line += 1
            col = 1
        else:
            col += 1
            
        if char == '{':
            stack.append(('{', line, col))
        elif char == '}':
            if not stack:
                print(f"Extra '}}' found at line {line}, col {col}")
                return
            stack.pop()
    
    if stack:
        for char, l, c in stack:
            print(f"Unclosed '{char}' from line {l}, col {c}")
    else:
        print("Brackets are balanced!")

if __name__ == "__main__":
    check_brackets(sys.argv[1])
